test/wallet-test.js

```diff
@@ -3273,4 +3273,141 @@ describe('Wallet', function() {
+        ),
+        bob: Buffer.from(
+          '097eca5a6d1a9ade3ef20ba3feab60b2dd1858f1b5888fb736917ec6ef0567e9',
+          'hex'
+        )
+      };
+      const expectedBlinds = {
+        alice: rules.blind(value, expectedNonces.alice),
+        bob: rules.blind(value, expectedNonces.bob)
+      };
+
+      // Initialize both multisig wallets
+      await alice.addSharedKey(0, await bob.accountKey(0));
+      await bob.addSharedKey(0, await alice.accountKey(0));
+
+      // P2SH address common for alice and bob
+      const addr = await alice.receiveAddress();
+      assert.strictEqual(
+        addr.toString(network),
+        'hs1q8nnrenk92qdmm32zfv56shvhj6n6g8kq3ermpz50k5zxh4qjq34qk7v4d2'
+      );
+
+      // Generate Nonce
+      // --------------
+
+      // Generate nonce for same inputs from both wallets
+      const aliceNonce = await alice.generateNonce(nameHash, addr, value);
+      const bobNonce = await bob.generateNonce(nameHash, addr, value);
+
+      // Same nonce is generated by alice and bob
+      // and is based on the smallest public key (alice's)
+      assert.bufferEqual(aliceNonce, expectedNonces.alice);
+      assert.bufferEqual(bobNonce, expectedNonces.alice);
+
+      // Generate nonces for all multisig participants
+      const aliceNonces = await alice.generateNonces(nameHash, addr, value);
+      const bobNonces = await bob.generateNonces(nameHash, addr, value);
+
+      // Both alice and bob get N nonces
+      assert.strictEqual(aliceNonces.length, 2);
+      assert.strictEqual(bobNonces.length, 2);
+
+      // and are the same, but ordered differently.
+      // First nonce is always based on own key,
+      // Then the rest in order of public key.
+      assert.deepStrictEqual(
+        aliceNonces,
+        [expectedNonces.alice, expectedNonces.bob]
+      );
+      assert.deepStrictEqual(
+        bobNonces,
+        [expectedNonces.bob, expectedNonces.alice]
+      );
+
+      // Generate Blind
+      // --------------
+
+      // sanity check: no blinds saved as of this point
+      assert(!await bob.txdb.hasBlind(expectedBlinds.alice));
+      assert(!await bob.txdb.hasBlind(expectedBlinds.bob));
+
+      // 1) Generate single blind:
+      const bobBlind = await bob.generateBlind(nameHash, addr, value);
+      // smallest public key (alice's) is used for blind
+      assert.bufferEqual(bobBlind, expectedBlinds.alice);
+      // bob's public key blind isn't stored
+      assert(await bob.txdb.hasBlind(expectedBlinds.alice));
+      assert(!await bob.txdb.hasBlind(expectedBlinds.bob));
+
+      // 2) Generate all blinds:
+      const bobBlinds = await bob.generateBlinds(nameHash, addr, value);
+      // own public key is used for blind (backwards-compatible)
+      assert.bufferEqual(bobBlinds[0], expectedBlinds.bob);
+      // but blinds for all key are saved in db
+      assert(await bob.txdb.hasBlind(expectedBlinds.alice));
+      assert(await bob.txdb.hasBlind(expectedBlinds.bob));
+    });
```

Comment:
I think you should also try to verify that both alice and bob can create a valid reveal even if the nonce is not based on their own key. That might even be best in `wallet-rpc-test.js` since thats the command thats been set up to cover the edge case.

Reply:
Added a multisig auction test in `wallet-rpc-test` from OPEN to CLOSED, including a failed no-blind reveal and importnonce that makes it work.